python/pyspark/ml/tests/test_util.py

```diff
@@ -73,6 +80,136 @@ def _check_uid_set_equal(stages, expected_stages):
         )
 
 
+class TestFunctionPickler(unittest.TestCase):
+
+
+    # Function that will be used to test pickling.
+    @staticmethod
+    def _test_function(x: float, y: float) -> float:
+        return x**2 + y**2
+
+    def _check_if_test_function_pickled(self, f, og_fn, output_value, *arguments, **key_word_args):
+        fn, args, kwargs = cloudpickle.load(f)
+        self.assertEqual(fn, og_fn)
+        self.assertEqual(args, arguments)
+        self.assertEqual(kwargs, key_word_args)
+        fn_output = fn(*args, **kwargs)
+        self.assertEqual(fn_output, output_value)
+
+    def test_pickle_fn_and_save(self):
+        x, y = 1, 3 # args of test_function
+        tmp_dir = "silly_goose"
+        os.mkdir(tmp_dir)
+        file_path_to_save = "silly_bear"
+
+        with self.subTest(msg="See if it pickles correctly if no file_path or save_dir are specified"):
+            pickled_fn_path = FunctionPickler.pickle_fn_and_save(TestFunctionPickler._test_function, "", "", x, y)
+            with open(pickled_fn_path, "rb") as f:
+                self._check_if_test_function_pickled(f, TestFunctionPickler._test_function, 10, x, y)
+            os.remove(pickled_fn_path)
+
+        with self.subTest(msg="See if pickles correctly and uses file path given as argument"):
+            pickled_fn_path = FunctionPickler.pickle_fn_and_save(TestFunctionPickler._test_function, file_path_to_save, "", x, y)
+            self.assertEqual(pickled_fn_path, file_path_to_save)
+            with open(pickled_fn_path, "rb") as f:
+                self._check_if_test_function_pickled(f, TestFunctionPickler._test_function, 10, x, y)
+            os.remove(pickled_fn_path)
+
+        with self.subTest(msg="See if pickles correctly and uses file path despite save_dir being specified"):
+            pickled_fn_path = FunctionPickler.pickle_fn_and_save(TestFunctionPickler._test_function, file_path_to_save, tmp_dir, x, y)
+            self.assertEqual(pickled_fn_path, file_path_to_save)
+            with open(pickled_fn_path, "rb") as f:
+                self._check_if_test_function_pickled(f, TestFunctionPickler._test_function, 10, x, y)
+            os.remove(pickled_fn_path)
```

Comment:
I don't think you need to test along these axes. The file and dir names shouldn't change the pickling behavior.

Reply:
I think this mainly tests if it saves it to the correct spot, which is why I included the tests. Do you think it'd be fine to keep to make sure this functionality works?

Reply:
I think it's unnecessary but it sounds fine